db/db.go

```diff
@@ -0,0 +1,129 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+
+	sq "github.com/Masterminds/squirrel"
+)
+
+// DB has all the tooling for talking to postgres
+type DB struct {
+	psql sq.StatementBuilderType
+}
+
+// New creates a DB instance
+func New(psql sq.StatementBuilderType) *DB {
+	return &DB{
+		psql: psql,
+	}
+}
+
+// Post represents a post in JSON
+type Post struct {
+	ID    int    `json:"id"`
+	Title string `json:"title"`
+	Body  string `json:"body"`
+}
+
+// GetPosts fetches all posts
+func (db *DB) GetPosts(ctx context.Context) (_ []Post, retErr error) {
+	posts := []Post{}
+
+	rows, err := db.psql.Select("*").From("posts").Query()
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		retErr = rows.Close()
```

Comment:
This will overwrite any other error returned, so I don't think this is what you want. You probably want to check whether retErr is nil first, or use something like https://github.com/uber-go/multierr to merge errors.
```suggestion
		err := rows.Close()
		if retErr == nil {
			retErr = err
		}
```

This sets retErr to the return value of `rows.Close` if it is nil.